### ISSUE TITLE
Error message complaining about build task

### ISSUE THREAD

**Issue by andreeis** [microsoft/vscode#176670]
The Makefile Tools extension creates a task when users want to build their code bases. The task uses "make" executable as command with arguments that we deduce from user settings.
This is how we write the task: [make.ts](https://github.com/microsoft/vscode-makefile-tools/blob/main/src/make.ts).
Around line 340. Do you spot anything wrong that we do there? So that is the typescript implementation of the Makefile Tools extension.

When we run the extension in VSCode and the user builds their code (which will hit the code I pointed above), the task appears to run fine (as it does for several releases already) but we see his message in "Tasks Output Window" that recently appeared.

```
Error: The task "makefile: Makefile Tools Build Task" has neither a command nor a dependsOn-property. The task will be ignored. Associated definition:
{
    "type": "shell",
    "id": "shell,c:\\amiga-gcc\\bin\\make.exe,all,-j4,DEBUG=1,",
    "problemMatcher": [
        "$gcc",
        "$unknown"
    ],
    "label": "makefile: Makefile Tools Build Task"
}

The problem matcher "{0}" can not be resolved. The matcher will be ignored.
```

Please advise, what can we do so that we don't see this message in the tasks window. Should we use the task API in a different way or this is a bug in your code base?

The bug report in our repository is https://github.com/microsoft/vscode-makefile-tools/issues/423 and there are many more duplicates that recently were signaled.

**Comment by andreeis** [comment]
No, I wasn't using insiders and either the developers in the community using the Makefile Tools extension. Also quite recent, a few weeks. But not sure exactly since when.

**Comment by meganrogge** [comment]
probably the same as #176496

**Comment by andreeis** [comment]
@meganrogge , I read the mentioned issue, just want to mention that in our case there is no debugger involved and also the problem matchers are not missing, they are defined in the package.json of our Makefile Tools extension (the one running when we see this error in the Tasks Output window).
Do you still need the "info needed" label?

**Comment by meganrogge** [comment]
Thanks for the info, this does look different. It looks like you pass args to be run but no `command`. Could you pass `make` as the `command`? 

```
{
    "type": "shell",
    "command": "make",
    "id": "shell,c:\\amiga-gcc\\bin\\make.exe,all,-j4,DEBUG=1,",
    "problemMatcher": [
        "$gcc",
        "$unknown"
    ],
    "label": "makefile: Makefile Tools Build Task"
}
```

<img width="871" alt="Screenshot 2023-03-10 at 1 34 11 PM" src="https://user-images.githubusercontent.com/29464607/224411253-c2c0db92-8ab0-463d-9c37-a9c7a87a365e.png">

**Comment by andreeis** [comment]
Please look at our source code, where we create the task: [make.ts](https://github.com/microsoft/vscode-makefile-tools/blob/main/src/make.ts).
I am not able to define a command directly into the task. I already define both command and args into the shell execution object that is passed in the task constructor.
Please point me to an example that is doing what you suggest but also look at what we are doing in the link I gave. Can you copy paste our fragment around line 345 and write the changes you suggest, how they would look like? I think I understand the "what" but it doesn't look like the APIs support that.

**Comment by andreeis** [comment]
@meganrogge , ping? Our users keep complaining about this error message in the tasks output window and we still don't know what to fix on our side :). Please advise.

**Comment by meganrogge** [comment]
nothing more from your side, thanks

**Comment by gcampbell-msft** [comment]
@meganrogge Following up on this issue. Any updates? Thank you very much, looking forward to hearing from you.

**Comment by NSExceptional** [comment]
@alexr00 replying to your comment from [here](https://github.com/microsoft/vscode/pull/200074#pullrequestreview-1769805734), there is more than one user experiencing this issue. Any time I create and start a `Task` like this, I get that warning in Ouptut:

```ts
const task = new Task(
    { type: 'shell' },
    TaskScope.Workspace,
    this.commandName,
    'TinderStudio',
    shellExecution,
    // problemMatchers
    [
        '$bazel-file',
        '$swift-lint',
        '$compiler-generic'
    ]
);

const taskExecution = await vscode.tasks.executeTask(task);
```

cc @meganrogge

**Comment by irenepc** [comment]
I just want to add a comment that I am seeing this issue on my end even AFTER having uninstalled the Makefile tools extension. It comes up whenever the command palette drops down with a list of issues, either activated from a keyboard shortcut or by going to Terminal > Run Task...
I submitted an issue for it here https://github.com/microsoft/vscode/issues/244042

**Comment by Provista-Stan** [comment]
+1 on this issue. Extensions like Microsoft Container tools are triggering this issue and it is very distracting to constantly dismiss this toast message that appears over terminal lists. An option to ignore errors of this type in the toast pop-up would be an improvement if this error cannot be completely removed.
